File: hw/tipantiza/u2/hw14MongoDBAndGUI/SafeStore_py/model/ui_adaptive.py
"""
Adaptive UI Module
@author Alexander Tipantiza, The Softwarriors, @ESPE
"""

class UIAdaptive:
    """Adaptive UI management"""
    
    _shortcuts = {}
    
    @classmethod
    def menu(cls):
        """Display adaptive UI menu"""
        print("\n--- ADAPTIVE UI ---")
        print("1. Apply theme")
        print("2. High contrast")
        print("3. Restore default")
        print("4. Assign shortcut")
        print("5. List shortcuts")
        print("0. Exit")
        return input("Option: ")
    
    @classmethod
    def execute_option(cls, option: str):
        """Execute selected option"""
        if option == "1":
            cls.apply_theme()
        elif option == "2":
            cls.high_contrast()
        elif option == "3":
            cls.restore_default()
        elif option == "4":
            cls.assign_shortcut()
        elif option == "5":
            cls.list_shortcuts()
        else:
            print("Invalid option")
    
    @classmethod
    def apply_theme(cls):
        """Apply UI theme"""
        theme = input("Enter theme name (dark/light): ")
        print(f"Theme '{theme}' applied")
    
    @classmethod
    def high_contrast(cls):
        """Enable high contrast mode"""
        print("High contrast activated")
    
    @classmethod
    def restore_default(cls):
        """Restore default UI settings"""
        cls._shortcuts.clear()
        print("Default values restored")
    
    @classmethod
    def assign_shortcut(cls):
        """Assign keyboard shortcut"""
        action = input("Enter action name: ")
        key = input("Enter shortcut key: ")
        cls._shortcuts[action] = key
        print(f"Shortcut assigned: {action} -> {key}")
    
    @classmethod
    def list_shortcuts(cls):
        """List all shortcuts"""
        if not cls._shortcuts:
            print("No shortcuts assigned")
        else:
            print("Current shortcuts:")
            for action, key in cls._shortcuts.items():
                print(f"  {action}: {key}")

File: hw/tipantiza/u2/hw14MongoDBAndGUI/SafeStore_java/src/main/java/ec/edu/espe/safestore/model/UIAdaptive.java

package ec.edu.espe.safestore.model;
import java.util.Scanner;
/**
 *
 * @author Alexander Tipantiza, The Softwarriors, @ESPE
 */


public class UIAdaptive {
    
    public static void menu(Scanner scanner) {
        System.out.println("\n--- ADAPTIVE UI ---");
        System.out.println("1. Apply theme");
        System.out.println("2. High contrast");
        System.out.println("3. Restore default");
        System.out.println("4. Assign shortcut");
        System.out.println("5. List shortcuts");
        System.out.print("Option: ");
        int option = scanner.nextInt();
        
        switch(option) {
            case 1: applyTheme(); break;
            case 2: highContrast(); break;
            case 3: restoreDefault(); break;
            case 4: assignShortcut(); break;
            case 5: listShortcuts(); break;
            default: System.out.println("Invalid option");
        }
    }
    
    private static void applyTheme() { System.out.println("Theme applied"); }
    private static void highContrast() { System.out.println("High contrast activated"); }
    private static void restoreDefault() { System.out.println("Default values restored"); }
    private static void assignShortcut() { System.out.println("Shortcut assigned"); }
    private static void listShortcuts() { System.out.println("Listing shortcuts"); }
}
